pkg/logging/logger.go

```diff
@@ -30,6 +31,7 @@ func (l LogLevel) String() string {
 }
 
 var defaultLogConsumer OptimizelyLogConsumer
+var mutex = &sync.Mutex{}
```

Comment:
Curious what issue was seen that we needed to protect the logger. The operations below seem to be atomic and shouldn't require a mutex, but I might be missing something.

Reply:
so, setLogger can be called at anytime.  During benchmarking, I was calling set at the begging and resetting at the end.  This caused a race condition which is now resolved.